tests/flytekit/unit/core/test_node_creation.py

```diff
@@ -166,19 +166,78 @@ def my_wf(a: int, b: str) -> (str, typing.List[str], int):
     @task
     def t1(a: str) -> str:
         return f"*~*~*~{a}*~*~*~"
 
     @workflow
     def my_wf(a: typing.List[str]) -> typing.List[str]:
         mappy = map_task(t1)
-        map_node = create_node(mappy, a=a).with_overrides(
+        map_node = mappy(a=a).with_overrides(requests=Resources(cpu="1", mem="100", ephemeral_storage="500Mi"))
+        return map_node
+
+    serialization_settings = context_manager.SerializationSettings(
+        project="test_proj",
+        domain="test_domain",
+        version="abc",
+        image_config=ImageConfig(Image(name="name", fqn="image", tag="name")),
+        env={},
+    )
+    wf_spec = get_serializable(OrderedDict(), serialization_settings, my_wf)
+    assert len(wf_spec.template.nodes) == 1
+    assert wf_spec.template.nodes[0].task_node.overrides is not None
+    assert wf_spec.template.nodes[0].task_node.overrides.resources.requests == [
+        _resources_models.ResourceEntry(_resources_models.ResourceName.CPU, "1"),
+        _resources_models.ResourceEntry(_resources_models.ResourceName.MEMORY, "100"),
+        _resources_models.ResourceEntry(_resources_models.ResourceName.EPHEMERAL_STORAGE, "500Mi"),
+    ]
+    assert wf_spec.template.nodes[0].task_node.overrides.resources.limits == []
+
+
+def test_resource_limits_override():
+    @task
+    def t1(a: str) -> str:
+        return f"*~*~*~{a}*~*~*~"
+
+    @workflow
+    def my_wf(a: typing.List[str]) -> typing.List[str]:
+        mappy = map_task(t1)
+        map_node = mappy(a=a).with_overrides(
+            limits=(Resources(cpu="2", mem="200", ephemeral_storage="1Gi")),
+        )
+        return map_node
+
+    serialization_settings = context_manager.SerializationSettings(
+        project="test_proj",
+        domain="test_domain",
+        version="abc",
+        image_config=ImageConfig(Image(name="name", fqn="image", tag="name")),
+        env={},
+    )
+    wf_spec = get_serializable(OrderedDict(), serialization_settings, my_wf)
+    assert len(wf_spec.template.nodes) == 1
+    assert wf_spec.template.nodes[0].task_node.overrides.resources.requests == []
+    assert wf_spec.template.nodes[0].task_node.overrides.resources.limits == [
+        _resources_models.ResourceEntry(_resources_models.ResourceName.CPU, "2"),
+        _resources_models.ResourceEntry(_resources_models.ResourceName.MEMORY, "200"),
+        _resources_models.ResourceEntry(_resources_models.ResourceName.EPHEMERAL_STORAGE, "1Gi"),
+    ]
+
+
+def test_resources_override():
+    @task
+    def t1(a: str) -> str:
+        return f"*~*~*~{a}*~*~*~"
+
+    @workflow
+    def my_wf(a: typing.List[str]) -> typing.List[str]:
+        mappy = map_task(t1)
+        map_node = mappy(a=a).with_overrides(
             requests=Resources(cpu="1", mem="100", ephemeral_storage="500Mi"),
-            limits=Resources(cpu="2", mem="200", ephemeral_storage="1Gi"),
+            limits=(Resources(cpu="2", mem="200", ephemeral_storage="1Gi")),
```

Comment:
Nit: you don't really need the extra `()` do you?

Reply:
Oof, nice catch. Will fix.